### DISCUSSION TITLE
Obsolete Field() functions

### DISCUSSION THREAD

**Discussion by Brilbroeder** [graphql-dotnet/graphql-dotnet#3659] [Q&A] [answered]
Hi,
we started using this nuget to support GraphQL in our application. We created a POC from some samples we found. POC was perfect. We used the version in the samples, we then upgraded to the latest nuget version and got the remark that ComplexGraphType.Field() is obsolete and being removed in the next version. Unfortunatly the documentation does not seems to be uptodate, or at least not clear to me :-)

In the constructor of a QueryI (public class RoleQuery : ObjectGraphType<object>) we have:

`Field<ListGraphType<RoleType>>("Roles", arguments: queryArguments, resolve: Resolve);`

Which uses the following:

```
 private List<Role> Resolve(IResolveFieldContext<object> context)
 {
}
```

Was is the correct 'long term' supported way to go ?

**Comment by Shane32** [comment, marked as answer]
See https://graphql-dotnet.github.io/docs/migrations/migration7/#11-a-bunch-of-fieldxxx-apis-were-deprecated
